thavalon-webapp/src/components/gameComponents/interactions/missionManager.tsx

```diff
@@ -102,50 +174,57 @@ export function MissionManager(props: MissionManagerProps): JSX.Element {
+                break;
+        }
         buttons.push(
             <button
+                className={className}
                 key={card}
                 onClick={() => props.submitMissionCard(card)}>
                 {card}
             </button>
         );
     }
     return (
-        <div className="mission-card-buttons">
+        <div className="mission-card-button-row">
             {buttons}
         </div>
     )
 }
 
 /**
- * Component showing the results of a mission with Agravaine support
- * @param props Required properties for the mission result modal
+ * Renders an appropriate after mission message for the player.
+ * For Agravaine, this will be a declare button or a message explaining why they
+ * can't declare. For others, this is a countdown for Agravaine.
+ * @param props Required properties for the AfterMissionMessage
  */
-export function MissionResultModal(props: MissionResultModalProps): JSX.Element {
-    if (props.message === undefined) {
-        return <></>;
-    }
-    const { successes, fails, reverses, questing_beasts, mission } = props.message;
+function AfterMissionMessage(props: AfterMissionMessageProps): JSX.Element {
+    const [timeToDeclare, setTimeToDeclare] = useState(AGRAVAINE_DECLARATION_TIME);
+    useEffect(() => {
+        const timer = setTimeout(() => setTimeToDeclare(timeToDeclare - 1), 1000);
+        return () => clearTimeout(timer);
+    });
+
     return (
-        <ReactModal
-            isOpen={true}
-            onRequestClose={() => props.setOpen(false)}
-            contentLabel="Mission Result Modal"
-            className="Modal"
-            overlayClassName="Overlay"
-        >
-            <div className="modalContainer">
-                <h2 className="modalHeader">Misison {mission} {props.message.passed ? "Passed!" : "Failed!"}</h2>
-                <hr />
-                <ul>
-                    <li key={"successes"}>Successes: {successes}</li>
-                    <li key={"fails"}>Fails: {fails}</li>
-                    <li key={"reverses"}>Reverses: {reverses}</li>
-                    <li key={"questing_beasts"}>Questing Beasts were here &lt;3: {questing_beasts}</li>
-                </ul>
-                <span className="waiting-for-agravaine">Waiting for Agravaine to declare...</span>
-                <button onClick={() => props.setOpen(false)}>Close</button>
-            </div>
-        </ReactModal>
+        <div className="after-mission-message">
+            {props.role !== Role.Agravaine && `Agravaine has ${ timeToDeclare } seconds to declare.`}
+            {props.role === Role.Agravaine && !props.onMission && "You cannot declare since you weren't on the mission. Do better."}
+            {props.role === Role.Agravaine && props.onMission &&
+                <button
+                    className="agravaine-declare-button"
+                    onClick={() => props.submitAgravaineDeclaration()}>
+                    Declare as Agravaine ({timeToDeclare}s)
```

Comment:
nit: this should say what the countdown is for, like `Declare as Agravaine ({timeToDeclare}s left)`